homeassistant/components/blebox/light.py

```diff
@@ -158,17 +158,16 @@ async def async_turn_on(self, **kwargs):
                 )
             else:
                 value = feature.apply_brightness(value, brightness)
-        if effect is not None:
-            effect_value = self.effect_list.index(effect)
+
+        try:
             await self._feature.async_on(value)
-            await self._feature.async_api_command("effect", effect_value)
-        else:
-            try:
-                await self._feature.async_on(value)
-            except ValueError as exc:
-                raise ValueError(
-                    f"Turning on '{self.name}' failed: Bad value {value}"
-                ) from exc
+            if effect is not None:
+                effect_value = self.effect_list.index(effect)
```

Comment:
If this can raise ValueError the re-raised exeption message will be confusing as `value` is not related to `effect`.

Reply:
Thanks for notice. Updated, effect set moved to separated `try` block, with adjusted message on possible error.
Added tests with different effect key values.